### PR TITLE
More sane behaviour when ES/Postgres servers are unavailable

### PR DESCRIPTION
**Pull request by 23Skidoo**
[Case 3116.](https://scrive.fogbugz.com/f/cases/3116/What-does-log-do-in-failure-cases-for-Postgres-and-Elasticsearch-backends)

### CODE REVIEW

**Comment by tageborg** [comment]
Gah. Misclick! Didn't mean to request review by phadej

File: log-base/src/Log/Logger.hs [hunk 1]
```diff
@@ -76,32 +77,48 @@ mkLogger name exec = mkLoggerImpl
...
-
--- | Check if an 'SQueue' is empty.
-isEmptySQueue :: SQueue a -> STM Bool
-isEmptySQueue (SQueue queue) = null <$> readTVar queue
-
--- | Read all the values stored in an 'SQueue'.
-readSQueue :: SQueue a -> STM [a]
-readSQueue (SQueue queue) = do
+-- | A simple STM based bounded queue.
+data SBQueue a = SBQueue !(TVar [a]) !(TVar Int) !Int
```
Comment (phadej) [inline log-base/src/Log/Logger.hs]: Why not `TBQueue` from  `stm`. If there's a good reason, maybe add a comment?

File: log-postgres/log-postgres.cabal [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name:                log-postgres
-version:             0.7
+version:             0.7.0.1
```
Comment (phadej) [inline log-postgres/log-postgres.cabal]: Do we need to bump version in `log` too, or could we deprecate the `log` package? re-exports are tricky from PVP pov.

File: log-base/src/Log/Logger.hs (hunk 1 above)
Comment (23Skidoo) [inline log-base/src/Log/Logger.hs]: Cause we need to be able to read all items in the queue with a single call.

File: log-postgres/log-postgres.cabal (hunk 2 above)
Comment (23Skidoo) [inline log-postgres/log-postgres.cabal]: I did bump it, because I added a benchmark, but the interface is the same.

**Comment by 23Skidoo** [comment]
OK, I'm gonna merge this and upload new releases to Hackage then.

**Pull request merged**